### ISSUE TITLE
Is not awaitable is incorrectly reported for `self` in classes with custom `__await__`

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#3464]
**Describe the bug**

```py
import asyncio
from typing import Generator, Any

class MyAwaitable:
    def __await__(self) -> Generator[Any, None, int]:
        async def foo() -> int:
            return 1

        return foo().__await__()

    async def foo(self) -> int:
        return await self  # error: "Self@MyAwaitable" is not awaitable

async def main() -> None:
    p = MyAwaitable()
    print(await p.foo())
    print(await p)

asyncio.run(main())
```

**Expected behavior**

No errors should be reported.

**VS Code extension or command-line**

Command-line `v1.1.247`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.248, which I just published. It will also be included in the next release of pylance.
